### PR TITLE
feat(fetch): use undici + `ProxyAgent` 

### PR DESCRIPTION
**Pull request by kanadgupta**
| 🚥 Resolves RM-8008, RM-9687 |
| :------------------- |

## 🧰 Changes

- [x] Removes `node-fetch` in favor of its native Node.js counterpart
- [x] Removes `formdata-node` in favor of its native Node.js counterpart
- [x] Uses undici's [`ProxyAgent`](https://undici.nodejs.org/#/docs/api/ProxyAgent) for proxying requests[^1]
- [x] Now that we installed the latest `nock` beta in #999, `msw` is no longer required and has been removed.
- [x] Removed a few useless `delayConnection` statements so now tests are like 2-3x faster

## 🧬 QA & Testing

Do tests still pass?

[^1]: We very well might have been doing proxying incorrectly this entire time? Ideally we should use the new [`EnvHttpProxyAgent`](https://github.com/nodejs/undici/blob/main/docs/docs/api/EnvHttpProxyAgent.md) API instead but that's only supported in `undici@6` 😔

### CODE REVIEW

File: __tests__/cmds/openapi/index.test.ts [hunk 1]
```diff
@@ -135,7 +133,6 @@ describe('rdme openapi', () => {
         .basicAuth({ user: key })
         .reply(200, [{ _id: 'spec1', title: 'spec1_title' }])
         .post('/api/v1/api-specification', { registryUUID })
-        .delayConnection(1000)
```
Comment (erunion) [inline __tests__/cmds/openapi/index.test.ts:138]: love to see it

File: __tests__/cmds/openapi/index.test.ts [hunk 2]
```diff
@@ -1350,37 +1330,24 @@ describe('rdme openapi', () => {
...
-      // TODO: move all of this boilerplate to the top-level once we migrate everything over to MSW
-      const server = setupServer(...[]);
+      const mock = getAPIMock()
+        .post('/api/v1/api-registry', body => body.match('form-data; name="spec"'))
+        .reply(201, { registryUUID });
 
-      server.listen({ onUnhandledRequest: 'error' });
+      const exampleMock = nock('https://example.com').get('/openapi.json').reply(200, petstoreWeird);
 
-      server.use(
```
Comment (erunion) [inline __tests__/cmds/openapi/index.test.ts:1362]: msw i think you're really neat but my dude... the code bloat

File: src/lib/readmeAPIFetch.ts [hunk 3]
```diff
@@ -182,15 +179,17 @@ export default async function readmeAPIFetch(
...
   debug(
-    `making ${(options.method || 'get').toUpperCase()} request to ${fullUrl} with headers: ${sanitizeHeaders(headers)}`,
+    `making ${(options.method || 'get').toUpperCase()} request to ${fullUrl} ${proxy ? `with proxy ${proxy} and ` : ''}with headers: ${sanitizeHeaders(headers)}`,
   );
 
-  return nodeFetch(fullUrl, {
+  return fetch(fullUrl, {
     ...options,
     headers,
+    dispatcher: proxy ? new ProxyAgent(proxy) : undefined,
```
Comment (erunion) [inline src/lib/readmeAPIFetch.ts:192]: lol @ the way node-fetch forced us to handle proxies 🤡

**Review by erunion — APPROVED** [review]

File: __tests__/cmds/openapi/index.test.ts (hunk 2 above)
Comment (kanadgupta) [reply __tests__/cmds/openapi/index.test.ts:1362]: yeah... once nock started using the same underlying interceptor, it was game over

File: src/lib/readmeAPIFetch.ts (hunk 3 above)
Comment (kanadgupta) [reply src/lib/readmeAPIFetch.ts:192]: lol that was not exactly a node-fetch issue and more of a me issue 😭

**Pull request merged**